packages/components/codemods/renameV2ComponentImportsAndUsages/renameV2ComponentImportsAndUsages.spec.ts

```diff
@@ -0,0 +1,390 @@
+import { parseJsx } from '../__tests__/utils'
+import {
+  getKaioTagNamesMapByComponentName,
+  printAst,
+  transformSource,
+  type TransformSourceArgs,
+} from '../utils'
+import { renameV2ComponentImportsAndUsages } from './renameV2ComponentImportsAndUsages'
+
+const transformComponents = (sourceFile: TransformSourceArgs['sourceFile']): string => {
+  const kaioTagNamesMap = getKaioTagNamesMapByComponentName(sourceFile, [
+    'Select',
+    'LikertScaleLegacy',
+    'TitleBlockZen',
+  ])
+  return transformSource({
+    sourceFile,
+    transformers: [renameV2ComponentImportsAndUsages(kaioTagNamesMap!)],
+  })
+}
+
+describe('renameV2ComponentImportsAndUsages', () => {
+  describe('Select -> SigleSelect', () => {
+    it('should rename Select import from next to SingleSelect', () => {
+      const inputAst = parseJsx(`import { Select } from "@kaizen/components/next"`)
+      const expectedAst = parseJsx(`import { SingleSelect } from "@kaizen/components"`)
+
+      const result = transformComponents(inputAst)
+      expect(result).toBe(printAst(expectedAst))
+    })
+
+    it('should rename Select import from future to SingleSelect', () => {
+      const inputAst = parseJsx(`import { Select } from "@kaizen/components/future"`)
+      const expectedAst = parseJsx(`import { SingleSelect } from "@kaizen/components"`)
+
+      const result = transformComponents(inputAst)
+      expect(result).toBe(printAst(expectedAst))
+    })
+
+    it('should rename Select JSX element to SingleSelect', () => {
+      const inputAst = parseJsx(`
+import { Select } from "@kaizen/components/next"
+
+const MyComponent = () => <Select />
+`)
+      const expectedAst = parseJsx(`
+import { SingleSelect } from "@kaizen/components"
+
+const MyComponent = () => <SingleSelect />
+`)
+
+      const result = transformComponents(inputAst)
+      expect(result).toBe(printAst(expectedAst))
+    })
+
+    it('should rename Select[Type] to SingleSelect[Type] from future module', () => {
+      const inputAst = parseJsx(`
+import type { SelectProps, SelectOption, SelectOptionGroup, SelectItem, SelectOptionNode, SelectOptionGroupNode, SelectItemNode } from "@kaizen/components/future"
+`)
+      const expectedAst = parseJsx(`
+import type { SingleSelectProps, SingleSelectOption, SingleSelectOptionGroup, SingleSelectItem, SingleSelectOptionNode, SingleSelectOptionGroupNode, SingleSelectItemNode } from "@kaizen/components"
+`)
+
+      const result = transformComponents(inputAst)
+      expect(result).toBe(printAst(expectedAst))
+    })
+
+    it('should rename Select[Type] to SingleSelect[Type] from next module', () => {
```

Comment:
This and the above test are technically already covered in the `updateKaioImports` tests I added too but this specifically covers all the type Props so thought it was worth doubling up here